doc/source/dev/releasing.rst

```diff
@@ -0,0 +1,210 @@
+with the Milestone for the release or the "backport-candidate" label.
+
+
+Creating the maintenance branch
+-------------------------------
+Before branching, ensure that the release notes are updated as far as possible.
+Include the output of ``tools/gh_lists.py`` and ``tools/authors.py`` in the
+release notes.
+
+Maintenance branches are named ``maintenance/0.<minor-version>.x``.  To create
+one, simply push a branch with the correct name to the scipy repo.  Immediately
+after, push a commit where you increment the version number on the master
+branch and add release notes for that new version.
+Send an email to scipy-dev to let people know that you've done this.
+
+
+Tagging a release
+-----------------
+First ensure that you have set up GPG correctly.  See
+https://github.com/scipy/scipy/issues/4919 for a discussion of signing release
+tags, and http://keyring.debian.org/creating-key.html for instructions on
+creating a GPG key if you do not have one.
+
+To make your key more readily identifiable as you, consider sending your key
+to public keyservers, with a command such as::
+
+    gpg --send-keys <yourkeyid>
+
+Check that all relevant commits are in the branch.  In particular, check issues
+and PRs under the Milestone for the release
+(https://github.com/scipy/scipy/milestones), PRs labeled "backport-candidate",
+and that the release notes are up-to-date and included in the html docs.
+
+Then edit ``setup.py`` to get the correct version number (set
+``ISRELEASED = True``) and commit it with a message like ``REL: set version to
+<version-number>``.  Don't push this commit to the Scipy repo yet.
+
+Finally tag the release locally with ``git tag -s <v0.x.y>`` (the ``-s`` ensures
+the tag is signed).  Continue with building release artifacts (next section).
+Only push the release commit and tag to the scipy repo once you have built the
+docs and Windows installers successfully.  After that push, also push a second
+commit which increment the version number and sets ``ISRELEASED`` to False
+again.
+
+
+Building release artifacts
+--------------------------
+Here is a complete list of artifacts created for a release:
+
+- source archives (``.tar.gz``, ``.zip`` and ``.tar.xz``)
+- Windows ``.exe`` installers
+- Binary wheels, for OS X only at the moment
+- Documentation (``html``, ``pdf``)
+- A ``README`` file
+- A ``Changelog`` file
+
+Everything except the OS X wheels can be built on Linux with
+https://github.com/numpy/numpy-vendor.  On Debian/Ubuntu it works out of the
+box, on other Linux flavors some extra dependencies may need to be installed.
+See the README file in the numpy-vendor repo for details.
+
+To start building, push your release branch and signed tag to your personal
+Scipy fork on Github, and then do:
+
+    vagrant up
+    fab vagrant prepare
+    vagrant ssh
+    cd repos/scipy  # clone from your personal Github fork if needed
+    git fetch origin
+    git co <release-tag>
+    paver release
```

Comment:
When running for the very first time, this fails first because of cython 0.20. When this is reinstalled, it fails to find blas/lapack. OK, `sudo apt-get install libatlas-base-dev` lets it proceed a bit further until

```
---> pavement.bdist_superpack
Application tried to create a window, but no driver could be loaded.
Make sure that your X server is running and that $DISPLAY is set correctly.
err:systray:initialize_systray Could not create tray window
Traceback (most recent call last):
  File "setup.py", line 260, in <module>
    setup_package()
  File "setup.py", line 248, in setup_package
    from numpy.distutils.core import setup
ImportError: No module named numpy.distutils.core
RESTART WINDOWS BUILD.  See gh-2709.
Traceback (most recent call last):
  File "setup.py", line 260, in <module>
    setup_package()
  File "setup.py", line 248, in setup_package
    from numpy.distutils.core import setup
ImportError: No module named numpy.distutils.core
```